### PR TITLE
Feature/preload sound effects

### PR DESCRIPTION
**Pull request by David-Hanna**
Sound effects will play without delay now, but as a requirement any sound effect to be played must be in the preload xml. Also added a check in ModelManager that a model isn't being preloaded twice since I did this for sound and went back to make it consistent for models as well. This commit also changes the music between the menu states and the racing state.

### CODE REVIEW

File: HeatStroke/AudioPlayer/AudioPlayer.cpp
Comment (bradcooper314) [inline HeatStroke/AudioPlayer/AudioPlayer.cpp:112]: Minor thing really, but the standard has been to use printf statements.

File: CS483/CS483/Kartaclysm/KartGame.cpp
Comment (bradcooper314) [inline CS483/CS483/Kartaclysm/KartGame.cpp:30]: If you already took the steps to preload sound effects, why isn't music also preloaded?

File: CS483/CS483/Kartaclysm/StateMachine/Gameplay/StateMainMenu.cpp
Comment (bradcooper314) [inline CS483/CS483/Kartaclysm/StateMachine/Gameplay/StateMainMenu.cpp:39]: Every call to PlayMusic should have a StopMusic before it. The player can always return to the main menu by finishing or quitting a race, so we can't assume that no previous music would be playing.

File: HeatStroke/Graphics/Model/ModelManager.cpp
Comment (bradcooper314) [inline HeatStroke/Graphics/Model/ModelManager.cpp]: It should be (it == m_mLoadedModels.end()). You want it to reach the end because you don't want it already in the list.

**Review by bradcooper314 — CHANGES_REQUESTED** [review]
The console says failure to load kingpin_turkey.wav.

Also needs develop merged into it before approval.

**Comment by bradcooper314** [comment]
Your sprint log also says music will stop while the race is paused. Easy addition to Suspend/Unsuspend.

**Comment by David-Hanna** [comment]
Something weird about that turkey sound effect file file because the path is correct and all the other sounds work. May just need a new sound effect for that one.

File: HeatStroke/AudioPlayer/AudioPlayer.cpp
Comment (David-Hanna) [reply HeatStroke/AudioPlayer/AudioPlayer.cpp:112]: I don't know about standard. I've been using std::cout since the beginning, and I was just not really bothering to correct people using printf.

File: CS483/CS483/Kartaclysm/KartGame.cpp
Comment (David-Hanna) [reply CS483/CS483/Kartaclysm/KartGame.cpp:30]: Because music is streamed. Loading a music file is huge, so the correct practice is simply to stream it instead.

File: HeatStroke/Graphics/Model/ModelManager.cpp
Comment (David-Hanna) [reply HeatStroke/Graphics/Model/ModelManager.cpp]: Silly mistake.

**Comment by David-Hanna** [comment]
The sprint log says it won't restart. Pausing the music during game pause wasn't something I thought was designed.

**Review by bradcooper314 — APPROVED** [review]
Still the issue with kingpin_turkey.wav, but it doesn't cause any errors during gameplay. Resolve conflicts and merge.

**Pull request merged**
